test/cache_test.rb

```diff
@@ -4,6 +4,26 @@
 
 module ActiveModelSerializers
   class CacheTest < ActiveSupport::TestCase
+    class AuthorWithCustomCacheKey < Author
+      class DerivedCacheKey
+        def initialize(object, method)
+          @object = object
+          @method = method
+        end
+
+        def cache_key
+          @object.__send__(@method)
+        end
+      end
+
+      def cache_key
+        [
+          DerivedCacheKey.new(self, :name),
+          DerivedCacheKey.new(self, :id)
+        ]
```

Comment:
@markiz my only real concern is that this class is really weird-looking. Would you mind adding some comments why you designed it this way? Be nice to future devs :)